src/Cli/dotnet/Commands/Run/RunCommand.cs

```diff
@@ -182,16 +202,85 @@ public int Execute()
         }
     }
 
-    internal void ApplyLaunchSettingsProfileToCommand(ICommand targetCommand, ProjectLaunchSettingsModel? launchSettings)
+    private int ExecuteWithExecutableProfile(ExecutableLaunchSettingsModel launchSettings)
+    {
+        Debug.Assert(launchSettings.ExecutablePath != null);
+
+        // Expand environment variables in the executable path
+        string executablePath = Environment.ExpandEnvironmentVariables(launchSettings.ExecutablePath);
+
+        // Determine working directory - use specified directory or project directory
+        string workingDirectory;
+        if (!string.IsNullOrEmpty(launchSettings.WorkingDirectory))
+        {
+            workingDirectory = Environment.ExpandEnvironmentVariables(launchSettings.WorkingDirectory);
+        }
+        else
+        {
+            // Default to the directory containing the project or entry point file
+            workingDirectory = Path.GetDirectoryName(ProjectFileFullPath ?? EntryPointFileFullPath ?? Directory.GetCurrentDirectory())!;
+        }
+
+        // Build command arguments by combining launch profile args with application args
+        // Note: launchSettings.CommandLineArgs is a pre-formatted string from JSON that cannot be reliably parsed
+        // ApplicationArgs are provided as separate tokens and need proper escaping
+        string commandArgs;
+        if (!NoLaunchProfileArguments && !string.IsNullOrEmpty(launchSettings.CommandLineArgs) && ApplicationArgs.Length > 0)
+        {
+            // Both launch profile args and application args - concatenate them
+            commandArgs = launchSettings.CommandLineArgs + " " + ArgumentEscaper.EscapeAndConcatenateArgArrayForProcessStart(ApplicationArgs);
+        }
+        else if (!NoLaunchProfileArguments && !string.IsNullOrEmpty(launchSettings.CommandLineArgs))
+        {
+            // Only launch profile args
+            commandArgs = launchSettings.CommandLineArgs;
+        }
+        else if (ApplicationArgs.Length > 0)
+        {
+            // Only application args
+            commandArgs = ArgumentEscaper.EscapeAndConcatenateArgArrayForProcessStart(ApplicationArgs);
+        }
+        else
+        {
+            commandArgs = string.Empty;
+        }
+
+        var commandSpec = new CommandSpec(executablePath, commandArgs);
+        var command = CommandFactoryUsingResolver.Create(commandSpec)
+            .WorkingDirectory(workingDirectory);
+
+        // Apply environment variables from launch profile
+        ApplyLaunchSettingsProfileToCommand(command, launchSettings);
```

Comment:
Seems like starting from here the code is the same as ExecuteWithProjectProfile.

Might be better to refactor these methods to methods that return a command based on the profile and then common code that sets env variables, sends telemetry and executes the command.